### PR TITLE
[opt](stats) Support display of auto analyze jobs

### PR DESCRIPTION
**Pull request by Kikyou1997**
### Support dispaly of auto analyze jobs

After this PR, users and DBA could use such grammar to check the execution status of auto analyze jobs:

```sql

SHOW AUTO ANALYZE [tbl_name] [WHERE STATE='SOME STATE']
```

Record count of history auto analyze job could be configured by setting FE option: auto_analyze_job_record_count, default value is 2000

### Enhance auto analyze

After this PR, auto jobs those created automatically will no longer execute beyond a specific time frame.

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
run buildall

File: fe/fe-common/src/main/java/org/apache/doris/common/Config.java [hunk 1]
```diff
@@ -2165,4 +2165,10 @@ public class Config extends ConfigBase {
                     + "The larger the value, the more uniform the distribution of the hash algorithm, "
                     + "but it will increase the memory overhead."})
     public static int virtual_node_number = 2048;
+
+    @ConfField(description = {
+            "控制系统自动触发作业执行记录的持久化行数",
```
Comment (morrySnow) [inline fe/fe-common/src/main/java/org/apache/doris/common/Config.java:2170]: In all descriptions, make it clear that these parameters control the collection of statistics

File: fe/fe-core/src/main/cup/sql_parser.cup [hunk 2]
```diff
@@ -4216,13 +4216,17 @@ show_param ::=
         RESULT = new ShowCreateMaterializedViewStmt(mvName, tableName);
     :}
     /* show analyze job */
-    | KW_ANALYZE opt_table_name:tbl opt_wild_where order_by_clause:orderByClause limit_clause:limitClause
```
Comment (morrySnow) [inline fe/fe-core/src/main/cup/sql_parser.cup:4219]: why remove order by and limit?

File: fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java [hunk 3]
```diff
@@ -2586,6 +2586,7 @@ public List<ResultRow> executeInternalQuery() {
                     analyze(context.getSessionVariable().toThrift());
                 }
             } catch (Exception e) {
+                LOG.warn("Failed to run internal SQL: {}", originStmt.toString(), e);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/qe/StmtExecutor.java:2589]:
```suggestion
                LOG.warn("Failed to run internal SQL: {}", originStmt, e);
```

File: fe/fe-core/src/main/java/org/apache/doris/statistics/AnalysisManager.java [hunk 4]
```diff
@@ -175,27 +183,44 @@ public class AnalysisManager extends Daemon implements Writable {
...
         if (job == null) {
             return null;
         }
-        for (BaseAnalysisTask task : analysisJobIdToTaskMap.get(info.jobId).values()) {
-            if (!task.info.state.equals(AnalysisState.FINISHED)) {
-                if (task.info.state.equals(AnalysisState.FAILED)) {
-                    systemJobInfoMap.remove(info.jobId);
-                }
+        int failedCount = 0;
+        StringJoiner reason = new StringJoiner(",");
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/statistics/AnalysisManager.java:197]:
```suggestion
        StringJoiner reason = new StringJoiner(", ");
```

File: fe/fe-core/src/main/cup/sql_parser.cup (hunk 2 above)
Comment (Kikyou1997) [reply fe/fe-core/src/main/cup/sql_parser.cup:4219]: 这是之前还有analyze表的时候遗留的代码, 对元数据结构实现这个操作比较复杂。且之前也不支持

**Comment by Kikyou1997** [comment]
run buildall

**Comment by Kikyou1997** [comment]
run buildall

**Review by morrySnow — DISMISSED** [review]

**Comment by Kikyou1997** [comment]
run buildall

**Comment by englefly** [comment]
Shall we expose job table schema to user? Let users create any filter condition they want.

**Review by morrySnow — APPROVED** [review]

**Comment by Kikyou1997** [comment]
> Shall we expose job table schema to user? Let users create any filter condition they want.

We will but not now

**Review by englefly — APPROVED** [review]

**Pull request merged**